src/middleware/bearer-auth/index.ts

```diff
@@ -77,24 +97,48 @@ export const bearerAuth = (options: BearerAuthOptions): MiddlewareHandler => {
     const headerToken = c.req.header(options.headerName || HEADER)
     if (!headerToken) {
       // No Authorization header
-      const res = new Response('Unauthorized', {
-        status: 401,
-        headers: {
-          'WWW-Authenticate': `${wwwAuthenticatePrefix}realm="` + realm + '"',
-        },
-      })
-      throw new HTTPException(401, { res })
+      const status = 401
+      const headers = {
+        'WWW-Authenticate': `${wwwAuthenticatePrefix}realm="` + realm + '"',
+      }
+      const responseMessage =
```

Comment:
The logic with these lines appears three times in this file. Can we make them into one function with unified logic?

Reply:
Absolutely, I will update the PR later today.

Reply:
@yusukebe PR updated.